cms/models.py

```diff
@@ -97,6 +97,30 @@ class CategorizedFaqsPage(Page):
     parent_page_types = ['FaqsPage']
 
 
+class GenericPage(Page):
```

Comment:
we can abstract the idea of `ProgramPage` children like this class and `FaqsPage`. it can be defined like this:

```python
class ProgramChildPage(Page):
    parent_page_types = ['ProgramPage']

    def parent_page(self):
        """ Get the parent ProgramPage"""
        return ProgramPage.objects.ancestor_of(self).first()

    def get_context(self, request):
        context = get_program_page_context(self.parent_page(), request)
        context['page'] = self
        context['active_tab'] = self.title
        return context
```

this class and `FaqPage` could inherit from that. note that `FaqPage` would have a `context['page']` key instead of a `context['faqpage']` key if you made this change. that's fine since it's just the name of a variable in the template.